concepts/eventing/performance evaluation/nats-jetstream/results_prod_HA.md

```diff
@@ -0,0 +1,329 @@
+       Active Consumers: 1
+  ```
+
+**Test Results:**
+```
+-> Total Events Sent      by **Test Sender** : 43,083 (+ 1 Failed = 43,084)
+-> Total Events Received  by **Sink**        : 43,012 (There were some errors in sink for storing the data to database)
+```
+
+**Test Results Dashboard:**
+
+![](assets/ha_crash4_1.png "")
+
+**Test Sender Stats:**
+
+![](assets/ha_crash4_2.png "")
+
+**Finding:** 
+
+As now there were 3 stream replicas, deleting the K8s Node where stream leader NATS Pod was deployed triggered a leader election for the stream. During the election time, event-publisher-proxy failed to publish some events to NATS. But this downtime is much less than the `Test Scenario 2` where there was only one stream replica and it had to wait for the same NATS Pod to come up in order to resume receiveing events. Instead, in this scenario a new NATS Pod was elected as the leader for the stream.
+
+### Run ID: 9/6/2022T9:22 (Duration: 5m, Event Rate: 150rps)
+
+> **NOTE:** Deleted k8s Node where `eventing-nats-2` Pod was deployed after 2 minutes of test execution. Eventing-controller pod was not deployed on the deleted node.
+
+**State before test run:**
+- Stream:
+  - Leader: eventing-nats-2
+  - Replicas: 3
+
+**State after test run:**
+- Stream:
+  - Leader: eventing-nats-1
+  - Replicas: 3
+- Consumer:
+  - Redelivered Messages: 0
+  - Unprocessed Messages: 29,386
+
+**Test Results:**
+```
+-> Total Events Sent      by **Test Sender** : 41,956 (+ 1 Failed = 41,957)
+-> Total Events Received  by **Sink**        : 41,960 (means 4 events were duplicates)
+```
+
+**Test Results Dashboard:**
+
+![](assets/ha_crash6_5.png "")
+
+**Test Sender Stats:**
+
+![](assets/ha_crash6_1.png "")
+
+**NATS JetStream Dashboard:**
+
+![](assets/ha_crash6_3.png "")
+
+**NATS Consumer Info:**
+
+![](assets/ha_crash6_2.png "")
+
+**NATS Stream Info:**
+
+![](assets/ha_crash6_4.png "")
+
+**Finding:** 
+
+After the test run, the consumer info showed `Unprocessed Messages: 29,386` but the `Redelivered Messages` and `Outstanding Acks` were zero. The eventing-controller was healthy so the unprocessed messages should have being delivered by NATS but they were not. Further investigation unveiled that there were some accounting bugs in NATS v2.8.2 and is fixed in new NATS v2.8.4 by [PR](https://github.com/nats-io/nats-server/pull/3148).
+
+## Conclusion
+Based on the findings of the test scenarios, it was seen that having a single stream replica does not ensure high availability as clustering in JetStream uses RAFT ([more info](https://docs.nats.io/running-a-nats-service/configuration/clustering/jetstream_clustering)). It is recommended to have atleast 3 stream replicas to provide high availibilty of stream in case of single-node failure. Therefore, now the production profile will use 3 replicas of stream and evaluation profile will still be set to use 1 stream replica ([related PR](https://github.com/kyma-project/kyma/pull/14539)).
```

Comment:
typos:
```suggestion
Based on the findings of the test scenarios, it was seen that having a single stream replica does not ensure high availability as clustering in JetStream uses RAFT ([more info](https://docs.nats.io/running-a-nats-service/configuration/clustering/jetstream_clustering)). It is recommended to have at least 3 stream replicas to provide high availability of a stream in case of single-node failure. Therefore, now the production profile will use 3 replicas of stream and evaluation profile will still be set to use 1 stream replica ([related PR](https://github.com/kyma-project/kyma/pull/14539)).
```